moveos/moveos-verifier/src/verifier.rs

```diff
@@ -2031,3 +2031,211 @@ impl TypeDepthFormula {
         Ok(TypeDepthFormula::normalize(formulas))
     }
 }
+
+/// Check the metadata compatibility between the new and old modules.
+pub fn check_metadata_compatibility(
+    old_module: &CompiledModule,
+    new_module: &CompiledModule,
+) -> VMResult<bool> {
+    let old_module_metadata = match get_metadata_from_compiled_module(old_module) {
+        None => {
+            return generate_vm_error(
+                ErrorCode::MALFORMED_METADATA,
+                "function should not return values".to_string(),
```

Comment:
The error message seems like not match the ErrorCode

Reply:
Fixed